### PR TITLE
gh-94018: Remove trailing spaces in _sanitize_windows_name

### PR DESCRIPTION
**Pull request by Rygone**
* Issue: gh-94018

### CODE REVIEW

**Comment by hauntsaninja** [comment]
You'll want to rename your PR to "gh-94018 [...]" instead of "hg-94018"

File: Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst
Comment (hauntsaninja) [inline Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst]:
```suggestion
Remove trailing spaces in :func:`zipfile._sanitize_windows_name`
```

File: Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst
Comment (zware) [inline Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst]: `_sanitize_windows_name` is not a public or documented function, so it should not be mentioned here.

```suggestion
:mod:`zipfile` will now remove trailing spaces from path components on extraction.
```

**Review by zware — CHANGES_REQUESTED** [review]
There should be one or more tests added for this.  See `Lib/test/test_zipfile.py`.

The change itself looks reasonable enough to me, but this should be reviewed by someone who is more versed in the inner workings of zip files before it is merged.

**Review by zware — APPROVED** [review]
LGTM, but I'm not comfortable merging without an OK from one of our `zipfile` maintainers.

File: Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst
Comment (dignissimus) [inline Misc/NEWS.d/next/Windows/2022-06-20-22-32-14.gh-issue-94018.bycC3A.rst]:
```suggestion
:mod:`zipfile` will now remove trailing spaces from path components when extracting files on Windows.
```

**Comment by ambv** [comment]
LGTM. Leading or trailing spaces (ASCII `0x20`) are indeed not allowed on Windows and those characters are silently removed by the OS:
https://docs.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/file-folder-name-whitespace-characters

**Pull request merged**
